maro/rl/algorithms/dqn.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from maro.rl.algorithms.abs_algorithm import AbsAlgorithm
+from maro.utils import clone
+
+
+class DQNHyperParams:
+    """Hyper-parameter set for the DQN algorithm.
+
+    Args:
+        num_actions (int): number of possible actions
+        reward_decay (float): reward decay as defined in standard RL terminology
+        num_training_rounds_per_target_replacement (int): number of training frequency of target model replacement
+        tau (float): soft update coefficient, e.g., target_model = tau * eval_model + (1-tau) * target_model
+    """
+    __slots__ = ["num_actions", "reward_decay", "num_training_rounds_per_target_replacement", "tau"]
+
+    def __init__(
+        self, num_actions: int, reward_decay: float, num_training_rounds_per_target_replacement: int, tau: float = 1.0
+    ):
+        self.num_actions = num_actions
+        self.reward_decay = reward_decay
+        self.num_training_rounds_per_target_replacement = num_training_rounds_per_target_replacement
```

Comment:
update_frequency

Reply:
technically this is the inverse of update frequency